File: algorithms/Codility-master/Codility-master/7_StacksQueues/Nesting.js
// 100%
function solution(S) {
    // write your code in JavaScript (Node.js 8.9.4)
    if (S.length == 0) return 1;
    let pairs = {'(':')'};
    let stack = [];
    for (let i=0; i< S.length; i ++){
        
        if (S.charAt(i) in pairs){
            
            stack.push(pairs[S.charAt(i)]);
        }
        
        if (stack.length != 0){
            if ((stack[stack.length-1]) == S.charAt(i)){
                stack.pop();
            }
        }else{
            return 0;
        }
    }
    if (stack.length == 0) return 1
    return 0;
        
}

File: algorithms/Codility-master/Codility-master/7_StacksQueues/Nesting.py
# 100%
def solution(S):
    # write your code in Python 3.6
    if len(S) == 0: return 1
    pairs = { '(':')'}
    stack = []
    for item in S:
        if item in pairs:
            stack.append(pairs[item])
        
        if len(stack)!=0:
            if stack[-1] == item:
                del stack[-1]
        else:
            return 0
    if len(stack)==0: return 1
    return 0